src/Workspaces/Core/Portable/Classification/SyntaxClassification/AbstractSyntaxClassificationService.Worker.cs

```diff
@@ -44,7 +45,7 @@ internal partial class AbstractSyntaxClassificationService
                 _getTokenClassifiers = getTokenClassifiers;
                 _semanticModel = semanticModel;
                 _syntaxTree = semanticModel.SyntaxTree;
-                _textSpan = textSpan;
+                _textSpanIntervalTree = new TextSpanIntervalTree(textSpans);
```

Comment:
>new TextSpanIntervalTree(textSpans); [](http://example.com/codeflow?start=40&length=36)

As mentioned earlier, I'm not sure of the costs of creating this. You probably should verify in a profile if this is expensiive, and if so, share the one used here and in the semantic classifier